backend/app/middleware.py

```diff
@@ -66,66 +66,21 @@ def middleware(request):
-        session every time, save the changes and set a session cookie or delete
-        the session cookie if the session has been emptied.
-        """
-        try:
-            accessed = request.session.accessed
-            modified = request.session.modified
-            empty = request.session.is_empty()
-        except AttributeError:
-            pass
-        else:
-            host = request.get_host().split(':')[0]
-            # First check if we need to delete this cookie.
-            # The session should be deleted only if the session is entirely empty
-            if settings.SESSION_COOKIE_NAME in request.COOKIES and empty:
-                response.delete_cookie(
-                    settings.SESSION_COOKIE_NAME,
-                    path=settings.SESSION_COOKIE_PATH,
-                    domain=host,
-                    samesite=settings.SESSION_COOKIE_SAMESITE,
-                )
-            else:
-                if accessed:
-                    patch_vary_headers(response, ('Cookie',))
-                if (modified or settings.SESSION_SAVE_EVERY_REQUEST) and not empty:
-                    if request.session.get_expire_at_browser_close():
-                        max_age = None
-                        expires = None
-                    else:
-                        max_age = request.session.get_expiry_age()
-                        expires_time = time.time() + max_age
-                        expires = http_date(expires_time)
-                    # Save the session data and refresh the client cookie.
-                    # Skip session save for 500 responses, refs #3881.
-                    if response.status_code != 500:
-                        try:
-                            request.session.save()
-                        except UpdateError:
-                            raise SuspiciousOperation(
-                                "The request's session was deleted before the "
-                                "request completed. The user may have logged "
-                                "out in a concurrent request, for example."
-                            )
-                        response.set_cookie(
-                            settings.SESSION_COOKIE_NAME,
-                            request.session.session_key, max_age=max_age,
-                            expires=expires, domain=host,
-                            path=settings.SESSION_COOKIE_PATH,
-                            secure=settings.SESSION_COOKIE_SECURE or None,
-                            httponly=settings.SESSION_COOKIE_HTTPONLY or None,
-                            samesite=settings.SESSION_COOKIE_SAMESITE,
-                        )
-        return response
+        session = request.session
+        if settings.SESSION_COOKIE_REFRESH_INTERVAL >= settings.SESSION_COOKIE_AGE:
+            return ValueError("SESSION_COOKIE_REFRESH_INTERVAL must be less than SESSION_COOKIE_AGE")
```

Comment:
I don't think it's helpful to return an error to the client for a server setting error, as the client can do nothing about it. I'd just remove this check assuming it's the system admin's responsibility to get the settings right (it is).

Reply:
fixed in [d368a16](https://github.com/TheSpaghettiDetective/obico-server/pull/829/commits/d368a16fe66a9e97da3865289c9b3aef50407f67)